tlspuffin/src/test_utils.rs

```diff
@@ -1,69 +1,251 @@
+        temp_dir: Option<TempDir>,
+    }
+
+    impl ParametersGuard {
+        pub fn build_options(&self) -> PutOptions {
+            let port = self.port.to_string();
+            let mut options: Vec<(&str, &str)> =
+                vec![("port", &port), ("prog", &self.prog), ("args", &self.args)];
+            if let Some(cwd) = &self.cwd {
+                options.push(("cwd", cwd));
+            }
+            options.into()
+        }
+    }
+
+    fn gen_certificate() -> (String, String, TempDir) {
+        let temp_dir = tempdir().unwrap();
+
+        let key = temp_dir.path().join("key.pem");
+        let key_path = key.as_os_str().to_str().unwrap();
+        let cert = temp_dir.path().join("cert.pem");
+        let cert_path = cert.as_os_str().to_str().unwrap();
+
+        let openssl_gen_cert_args = [
+            "req",
+            "-x509",
+            "-newkey",
+            "rsa:2048",
+            "-keyout",
+            key_path,
+            "-out",
+            cert_path,
+            "-days",
+            "365",
+            "-nodes",
+            "-subj",
+            "/C=US/ST=New Sweden/L=Stockholm/O=.../OU=.../CN=.../emailAddress=...",
+        ];
+
+        let cert_output = collect_output(execute_command::<_, _, &str>(
+            OPENSSL_PROG,
+            openssl_gen_cert_args,
+            None,
+        ));
+        println!("Certificate generation: {}", cert_output);
+
+        (key_path.to_owned(), cert_path.to_owned(), temp_dir)
+    }
+
+    pub fn wolfssl_client(port: u16, version: TLSVersion, warmups: Option<u32>) -> ParametersGuard {
+        let (_key, _cert, temp_dir) = gen_certificate();
+
+        let port_string = port.to_string();
+        let mut args = vec!["-h", "127.0.0.1", "-p", &port_string, "-x", "-d"];
+        let prog = "./examples/client/client";
+        let cwd = "/home/max/projects/wolfssl";
+
+        match version {
+            TLSVersion::V1_3 => {
+                args.push("-v");
+                args.push("4");
+            }
+            TLSVersion::V1_2 => {
+                args.push("-v");
+                args.push("3");
+            }
+        }
+
+        let warmups = warmups.map(|warmups| warmups.to_string());
+
+        if let Some(warmups) = &warmups {
+            args.push("-b");
+            args.push(warmups);
+        }
+
+        ParametersGuard {
+            port,
+            prog: prog.to_owned(),
+            args: args.join(" "),
+            cwd: Some(cwd.to_owned()),
+            temp_dir: Some(temp_dir),
+        }
+    }
+
+    pub fn wolfssl_server(port: u16, version: TLSVersion) -> ParametersGuard {
+        let (_key, _cert, temp_dir) = gen_certificate();
+
+        let port_string = port.to_string();
+        let mut args = vec!["-p", &port_string, "-x", "-d", "-i"];
+        let prog = "./examples/server/server";
+        let cwd = "/home/max/projects/wolfssl";
```

Comment:
Maybe remove this absolute path

Reply:
We should definitely do something about this. I noticed several occurrences, so I opened #335 to take the time and evaluate what we should do with these in a future PR.